test/testdata/infer/generics/isa_with_type_member.rb

```diff
@@ -0,0 +1,99 @@
+# typed: true
+
+class A
+  extend T::Generic
+  X = type_member
+end
+
+class B < A
```

Comment:
Slight preference for calling these `Parent` / `Child` so that it's easier to see what the meaning is when glancing at any individual test case.

Reply:
I like that too.